lib/awful/tag.lua

```diff
@@ -240,11 +240,20 @@ end
 -- @return A table with all created tags.
 function tag.new(names, screen, layout)
     screen = get_screen(screen or 1)
-    local tags = {}
+
+    if layout and layout[1] and #names > #layout then
+        gdebug.print_warning("The number of tags is greater than the number of layouts")
+    end
+
+    local tags, overflow = {}, false
     for id, name in ipairs(names) do
-        table.insert(tags, id, tag.add(name, {screen = screen,
-                                            layout = (layout and layout[id]) or
-                                                        layout}))
+        table.insert(tags, id, tag.add(name, {
+            screen = screen,
+            layout = (layout and layout[id]) or (overflow and layout[1] or layout)
+        }))
+
+        overflow = overflow or (layout and layout[id])
```

Comment:
This `overflow` logic confuses me. So `layout` should always be trueish (if someone passes in `nil` or `false` here, then... things would work. Oh boy.).

New attempt:
If `layout` is `nil`, then all created tags end up without a layout. Good.
If `layout` is a `layout` without `layout[1]`, then... `(layout and layout[id])` would evaluate to `nil` and `(overflow and layout[1] or layout)` would evaluate to `layout`. Good, but complicated.
If `layout` is a sufficiently long list of layouts, then the first case in the tag-creation always fires and `overflow` stays falsish.
If `layout` has too few layouts... let's say `names = { "foo", "bar", "baz" }` and `layouts = { }`, then the first created tag will end up with... an empty table as its layout.... right? `layout and layout[id]` is `nil`, `overflow` is `false` and so the second part evaluates to `layout`.

So this patch is (a) too complicated and (b) does not fix things, I think.